internal/transport/keepalive_test.go

```diff
@@ -314,13 +314,13 @@ func (s) TestKeepaliveClientClosesWithActiveStreams(t *testing.T) {
 	copts := ConnectOptions{
 		ChannelzParentID: channelz.NewIdentifierForTesting(channelz.RefSubChannel, time.Now().Unix(), nil),
 		KeepaliveParams: keepalive.ClientParameters{
-			Time:    500 * time.Millisecond,
-			Timeout: 500 * time.Millisecond,
+			Time:    10 * time.Millisecond,
+			Timeout: 10 * time.Millisecond,
 		},
 	}
-	// TODO(i/6099): Setup a server which can ping and no-ping based on a flag to
-	// reduce the flakiness in this test.
-	client, cancel := setUpWithNoPingServer(t, copts, connCh)
+	respModeCh := make(chan bool, 1)
+	defer close(respModeCh)
+	client, cancel := setUpWithNoPingServerWithOptions(t, copts, connCh, respModeCh)
```

Comment:
I think it would be a small amount more work, but much nicer to make `setUpWithNoPingServer` return a `server` type that has a method that allows you to change the mode.  And that can take the place of the `cancel` return parameter with a `server.stop()` method instead.

Reply:
I didn't want to change the original pattern of these utility functions, but sure thing, I can look into that when I get a chance.

Reply:
I had a chance to look into this suggestion, and made some changes. I'll submit it later for your review. Thanks